src/sherpa_ai/agents/base.py

```diff
@@ -119,35 +125,85 @@ def validate_output(self):
         Returns:
             str: The synthesized output after validation.
         """
-        failed_validation = []
-        result = self.synthesize_output()
+        result = ""
+        # create array of instance of validation so that we can keep track of how many times regeneration happened.
+        instantiated_validations = []
         for validation in self.validations:
-            for count in range(self.validation_steps):
-                self.belief.update_internal(EventType.result, self.name, result)
+            try:
+                instantiated_validations.append(validation())
+            except Exception as e:
+                instantiated_validations.append(validation)
+
+        all_pass = False
+        validation_is_scaped = False
+        while_count = 0
+
+        # this loop will run until max regeneration reached or all validations have failed
+        while (
+            self.global_regen_max > sum(val.count for val in instantiated_validations)
+            and not all_pass
+        ):
+            logger.info(f"validations_size: {len(instantiated_validations)}")
```

Comment:
This part is getting over complicated, we should extract one iteration of validation into another function